### PR TITLE
Documentation updates. Adding some developer centric docs

### PR DESCRIPTION
**Pull request by analysis230**
There is no developer reference for the code at all. This is a small start to that. Looking into more ways of code base documentation to make things more comprehensive.

### CODE REVIEW

File: doc/developer-reference/Preset-and-bundle.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+This is a bundle containing a few types of `PresetCollection`. One bundle has presets for some printers, filaments and some processes (TYPE_PRINT).
+
+`PresetCollection            prints`\
+`PresetCollection            filaments`\
+`PrinterPresetCollection     printers`
+
+each one of these contains a collection of processes, filaments and printers respectively.\
+
+#### Clarification Needed: It is unclear to me if `printers` contain different nozzle sizes of the same printer model or all different printers models by the same vendor or something else. Feel free to add clarification.
```
Comment (Ocraftyone) [inline doc/developer-reference/Preset-and-bundle.md:33]: Each nozzle size is a different preset

File: doc/developer-reference/plater-sidebar-tab-combobox.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+## [`ComboBox`](../../src/slic3r/GUI/Widgets/ComboBox.hpp)
+
+The drop down menus where you can see and select presets
+
+<img src="../images/combobox.png" alt="Example Image" width="320">
+
+## [`Tab`](../../src/slic3r/GUI/Tab.hpp)
+
+Refers to the pop windows that open when you click the edit preset button
```
[the remark this answers is not in the document]
Comment (analysis230) [reply doc/developer-reference/plater-sidebar-tab-combobox.md]: you are right. I corrected that. Also realized that just like `TabPrint`, `TabPrintModel` is also not a popup.

File: doc/developer-reference/Preset-and-bundle.md (hunk 1 above)
Comment (analysis230) [reply doc/developer-reference/Preset-and-bundle.md]: Yes. I also understood that PresetBundle is just a bundle of whatever presets without regard for compatibility with each other. Compatibility is taken care of later while loading the presets to the UI.

Comment (Ocraftyone) [reply doc/developer-reference/Preset-and-bundle.md]: Sorry. My previous answer wasn't very clear. The printers preset bundle contains all printer presets. It isn't sorted by nozzle size or vendor.

Comment (analysis230) [reply doc/developer-reference/Preset-and-bundle.md]: yes. I also realized that while digging through the code earlier today. I have already made that update.

Comment (akoolenbourke) [reply doc/developer-reference/Preset-and-bundle.md]: > Each nozzle size is a different preset

I want to change this and so many other things if I'm honest. As someone relatively new to 3d printing the way slicers deal with their presets and poor override systems is really frustrating. I assume it's historic but I want one preset where some of the values will change if I just select a different nozzle. 

 I want to be able to say things like if this is PETG then just add some z offset. All my PETG has the same Z_ADJUST custom code.

**Review by SoftFever — APPROVED** [review]
This is fantastic first step of documenting the dev stuff.
Thank you!

**Pull request merged**